tensorflow/core/kernels/crop_and_resize_op.cc

```diff
@@ -442,70 +455,96 @@ struct CropAndResizeBackpropImage<CPUDevice, T> {
-                  static_cast<T>(x_lerp * dtop);
-              const float dbottom = y_lerp * grads(b, y, x, d);
-              grads_image(b_in, bottom_y_index, left_x_index, d) +=
-                  static_cast<T>((1 - x_lerp) * dbottom);
-              grads_image(b_in, bottom_y_index, right_x_index, d) +=
-                  static_cast<T>(x_lerp * dbottom);
+          const int closest_y_index = roundf(in_y);
+          const int top_y_index = floorf(in_y);
+          const int bottom_y_index = ceilf(in_y);
+          const float y_lerp = in_y - top_y_index;
+          const float one_y_lerp = 1 - y_lerp;
+
+          for (int x = 0; x < crop_width; ++x) {
+            const float in_x = (crop_width > 1)
+                                   ? x1 * (image_width - 1) + x * width_scale
+                                   : 0.5 * (x1 + x2) * (image_width - 1);
+            if (in_x < 0 || in_x > image_width - 1) {
+              continue;
             }
-          } else {  // method_name == "nearest"
-            for (int d = 0; d < depth; ++d) {
-              int closest_x_index = roundf(in_x);
-              int closest_y_index = roundf(in_y);
-              grads_image(b_in, closest_y_index, closest_x_index, d) +=
-                  static_cast<T>(grads(b, y, x, d));
+            const int closest_x_index = roundf(in_x);
+
+            if (isBilinear) {
+              const int left_x_index = floorf(in_x);
+              const int right_x_index = ceilf(in_x);
+              const float x_lerp = in_x - left_x_index;
+              const float one_x_lerp = 1 - x_lerp;
+
+              for (int d = 0; d < depth; ++d) {
+                const float dtop = one_y_lerp * grads(b, y, x, d);
+                grads_image(b_in, top_y_index, left_x_index, d) +=
+                    static_cast<T>(one_x_lerp * dtop);
+                grads_image(b_in, top_y_index, right_x_index, d) +=
+                    static_cast<T>(x_lerp * dtop);
+                const float dbottom = y_lerp * grads(b, y, x, d);
+                grads_image(b_in, bottom_y_index, left_x_index, d) +=
+                    static_cast<T>(one_x_lerp * dbottom);
+                grads_image(b_in, bottom_y_index, right_x_index, d) +=
+                    static_cast<T>(x_lerp * dbottom);
+              }
+            } else {  // method_name == "nearest"
+              for (int d = 0; d < depth; ++d) {
+                grads_image(b_in, closest_y_index, closest_x_index, d) +=
+                    static_cast<T>(grads(b, y, x, d));
+              }
             }
           }
         }
       }
-    }
+    };
+
+    // A rough estimation of the cost for each cropped box.
+    // Including calculation cost in the depth loop and pixel loop.
+    const double cost_per_pixel =
+        (isBilinear
+             ? depth * (Eigen::TensorOpCost::AddCost<float>() * 4 +
+                        Eigen::TensorOpCost::MulCost<float>() * 4 +
```

Comment:
I see 6 multiplications in there?
```suggestion
                        Eigen::TensorOpCost::MulCost<float>() * 6 +
```

Reply:
Done.